pyomo/core/expr/expr_pyomo5.py

```diff
@@ -1817,6 +1817,17 @@ def _compute_polynomial_degree(self, result):
     def getname(self, *args, **kwds):
         return 'prod'
 
+    def _is_fixed(self, args):
+        # Anything times 0 equals 0, so one of the children is
+        # fixed and has a value of 0, then this expression is fixed
+        assert(len(args) == 2)
+        if all(args):
+            return True
+        for i in (0, 1):
+            if args[i] and value(self._args_[i]) == 0:
+                return True
```

Comment:
It does seem like undefined behavior, but if `<value>` throws an exception, I think we can assume by convention that the value is not `<0>`. In this case, I could replace `<value(self._args_[i])>` with `<value(self._args_[i], exception=False)>` in `<_isfixed>` for both `<PowExpression>` and `<ProductExpression>`. This would cause exceptions to work like any other non 0 value. How does that sound?

Reply:
Per my previous note, I think we should defer this error checking.  I think that this is a broader issue for expression manipulation and management.

Reply:
I get the feeling that the solution will probably be "figure out what to do on a case by case basis". That being said, I am fine with this pull request sitting around until there is a consensus on what to do project wide.